compiler/compile/OSRData.hpp

```diff
@@ -91,7 +91,7 @@ namespace TR
       };
    }
 
-typedef TR::typed_allocator<std::pair<int32_t , TR_BitVector*>, TR::Region&> DefiningMapAllocator;
+typedef TR::typed_allocator<std::pair<const int32_t, TR_BitVector*>, TR::Region&> DefiningMapAllocator;
```

Comment:
Should all `std::map` keys be defined as `const` as a good rule of thumb? I have a PR in flight with a similar change.

Reply:
@fjeremic yes, otherwise I think you'll generate another example of the same problem this PR fixes :(

Reply:
(thanks for noticing :) )

Reply:
note that the std::map key type is still not const, though.... it's only on the allocator that you need the 'const'. If you do the key type on DefiningMap, for example, that will break on AIX and zOS (as I found with my earlier version of this fix). Unfortunately, we do not currently have compiler component testing on those platforms at Eclipse OMR (we're working on it!)

Reply:
Makes sense given the definition of the allocator from the documentation (http://en.cppreference.com/w/cpp/container/map). Thanks! I'll make the same change accordingly.